code/src/test/kotlin/com/expediagroup/sdk/core/http/MediaTypeTest.kt

```diff
@@ -0,0 +1,545 @@
+    @Test
+    fun `should create MediaType with provided parameters`() {
+        // Given
+        val mediaType = MediaType.of("text", "html", mapOf("charset" to "UTF-8"))
+
+        // When
+        val type = mediaType.type
+        val subtype = mediaType.subtype
+        val parameters = mediaType.parameters
+
+        // Expect
+        assertEquals("text", type)
+        assertEquals("html", subtype)
+        assertEquals(mapOf("charset" to "UTF-8"), parameters)
+    }
+
+    @Test
+    fun `should normalize type, subtype, and parameters to lowercase`() {
+        // Given
+        val mediaType = MediaType.of("APPLICATION", "JSON", mapOf("CHARSET" to "UTF-8"))
+
+        // When
+        val type = mediaType.type
+        val subtype = mediaType.subtype
+        val parameters = mediaType.parameters
+
+        // Expect
+        assertEquals("application", type)
+        assertEquals("json", subtype)
+        assertEquals(mapOf("charset" to "UTF-8"), parameters)
+    }
+
+
+    @Test
+    fun `should throw an exception for blank type`() {
+        // Given
+        val blankType = ""
+
+        // When
+        val exception = assertThrows<IllegalArgumentException> {
+            MediaType.of(blankType, "json")
+        }
+
+        // Expect
+        assertEquals("Type must not be blank", exception.message)
+    }
+
+    @Test
+    fun `should throw an exception for black subtype`() {
+        // Given
+        val blankSubType = ""
+
+        // When
+        val exception = assertThrows<IllegalArgumentException> {
+            MediaType.of("application", blankSubType)
+        }
+
+        // Expect
+        assertEquals("Subtype must not be blank", exception.message)
+    }
+
+    @Test
+    fun `should return full type`() {
+        // Given
+        val mediaType = MediaType.of("application", "json")
+
+        // When
+        val fullType = mediaType.fullType
+
+        // Expect
+        assertEquals("application/json", fullType)
+    }
+
+    @Test
+    fun `should return charset when present`() {
+        // Given
+        val mediaType = MediaType.of("text", "html", mapOf("charset" to "UTF-8"))
+
+        // When
+        val charset = mediaType.charset
+
+        // Expect
+        assertEquals(Charset.forName("UTF-8"), charset)
+    }
+
+    @Test
+    fun `should return null charset when not present`() {
+        // Given
+        val mediaType = MediaType.of("text", "html")
+
+        // When
+        val charset = mediaType.charset
+
+        // Expect
+        assertNull(charset)
+    }
+
+    @Test
+    fun `should return null charset for invalid charset`() {
+        // Given
+        val mediaType = MediaType.of("text", "html", mapOf("charset" to "INVALID"))
+
+        // When
+        val charset = mediaType.charset
+
+        // Expect
+        assertNull(charset)
+    }
+
+    @Test
+    fun `should include media type with wildcard type`() {
+        // Given
+        val wildcardType = MediaType.of("*", "json")
```

Comment:
Is a defined subtype with a wildcard type valid?

Reply:
shouldn't, at least logically...

Reply:
addressed the case https://github.com/ExpediaGroup/lodging-connectivity-java-sdk/pull/117/commits/4bb6b78325ba82f76dbced30a2b055a044fcd701